package.json

```diff
@@ -2,9 +2,12 @@
   "name": "tsyringe",
   "version": "2.2.0",
   "description": "Lightweight dependency injection container for JavaScript/TypeScript",
-  "main": "dist/index.js",
+  "main": "dist/cjs/index.js",
+  "module": "./dist/es5/index.js",
+  "es2015": "./dist/es2015/index.js",
+  "typings": "./dist/es2015/index.d.ts",
```

Comment:
I don't think so and only changed it to use es2015 to test. I'll review how other projects handle it.

Reply:
I added tsconfig.types.json and added another build step. I'm still checking how other projects handle it but that is how rxjs handles it.

Reply:
Could you provide documentation on the support for different module types? And a couple of usage examples in the README.md?

Reply:
Yeah, I'll add a note about what each of the main files (builds) is for. For usage examples, the bundle libraries handle it and the implementation is no different than using cjs.

Reply:
Cool. I assume no one will even hit this without using a bundler? (Asking since I always use cjs and let the bundler convert)

Reply:
So reading the (very sparse) [Webpack documentation](https://webpack.js.org/guides/author-libraries/#final-steps) and its links, it seems that `module` should point to the `es2015` version; does this read the same to you also?

Reply:
The documentation says `ES2015 modules` which is what the `module` and `es2015` main fields use. Rollup and Webpack only care about the module format when bundling. 

Creating the ES5 and ES2015 builds gives consumers options depending on the browser environment they need to support. Based on everything I've read, the way I set it up in this PR is what most are using.

It follows how Angular does it (they include more but same idea):
https://docs.google.com/document/d/1CZC2rcpxffTDfRDs6p1cfbmKNLA6x5O-NtkJglDaBVs/preview
https://github.com/angular/angular/blob/master/packages/core/package.json#L9

Reply:
> Cool. I assume no one will even hit this without using a bundler? (Asking since I always use cjs and let the bundler convert)

Once Node supports ES Modules without a command line flag, it will be used there too.